src/global/project/manifest.rs

```diff
@@ -431,40 +431,52 @@ impl Manifest {
 #[derive(Debug, Clone, Serialize, Deserialize, Hash, PartialEq, Eq)]
 pub struct Mapping {
     exposed_name: ExposedName,
-    executable_name: String,
+    executable_relname: String,
 }
 
 impl Mapping {
-    pub fn new(exposed_name: ExposedName, executable_name: String) -> Self {
+    pub fn new(exposed_name: ExposedName, executable_relname: String) -> Self {
         Self {
             exposed_name,
-            executable_name,
+            executable_relname: Path::new(&executable_relname)
+                .with_extension("")
+                .to_str()
+                .unwrap_or(&executable_relname)
+                .to_string(),
         }
     }
 
     pub fn exposed_name(&self) -> &ExposedName {
         &self.exposed_name
     }
 
+    pub fn executable_relname(&self) -> &str {
+        &self.executable_relname
+    }
     pub fn executable_name(&self) -> &str {
-        &self.executable_name
+        if let Some(executable_file_name) = Path::new(&self.executable_relname).file_name() {
+            return executable_file_name
+                .to_str()
+                .unwrap_or(&self.executable_relname);
+        };
+        &self.executable_relname
```

Comment:
I believe you can use the `pixi_utils::executable_from_path` again

Reply:
I could have use it but it returns a String not a &str, it would break many things.

Reply:
Here I'm not trimming the extension, self.executable_relname is already trimmed at instantiation (in fn new) and file_name() will not trim python3.12
I can't use .as_str() because it would be a reference to a temporary value created inside the method that won't persist.
If you want I could change executable_from_path and strip_executable_extension to take and return &str but It might be a bit overkill.
What do you prefer?

Reply:
I'm fine with it, I cleaned it up a little more in my diff.